### PR TITLE
v0.4.0 Release Candidate

### PR DESCRIPTION
**Pull request by gzagaris**
# Summary

v0.4.0 release candidate branch

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -18,7 +142,11 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 - Added [CLI11](https://github.com/CLIUtils/CLI11) command line parser as a built-in third party library.
 
 ### Removed
-
+- Removed `AXOM_ENABLE_CUB` option, since `CUB` is no londer used directly in
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:145]: Typo: `londer` -> `longer`

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -7,6 +7,130 @@ The format of this file is based on [Keep a Changelog](http://keepachangelog.com
...
+- Third-party libraries can now build on the Windows platform through uberenv using vcpkg
+  ("zero-to-axom support on Windows")
+
+### Known Issues
+- Encountered a compiler bug on IBM LC platforms when using the IBM XL C/C++
+  compiler. The issue is manifested in the `generate_aabbs_and_centroids` method
+  in the `spin_bvh.cpp` unit test. It seems that the compiler does not handle
+  the lambda capture of the arrays correctly which leads to a segfault. A
+  workaround for the IBM XL compiler is provided.
+- There is known bug in MVAPICH that prevents consecutive creation/deletion
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:129]: Typo: 
>There is \*a\* known bug....

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -7,6 +7,130 @@ The format of this file is based on [Keep a Changelog](http://keepachangelog.com
...
+- Added option to quest's `SignedDistance` class and C API to toggle whether
+  the distance query computes the sign.
+- Added a `batched` option to quest's signed distance query example application.
+  This computes all distance queries on an array of points using a single call to `computeDistance`.
+  The query uses OpenMP threading, when available.
+- Added new component, Inlet, to assist in retrieving and storing data from
+  an input deck.
+- Added the ability to specify an [Umpire] allocator ID to use with the
+  BVH. This allows the application to use a device allocator for the BVH and 
+  avoid use of UM on the GPU, which can hinder perfomrmance, or use a pool
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:30]: Suggestion: UM -> "Unified Memory (UM)" ?  (in case a reader is not familiar with that acronym)

**Review by kennyweiss — APPROVED** [review]
Thanks @gzagaris

**Comment by gzagaris** [comment]
Folks, at your convenience, please take a look at this PR. There is a user that is currently waiting on the v0.4.0 release.

File: scripts/uberenv/packages/axom/package.py [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+       of multi-physics applications and computational tools."""
+
+    maintainers = ['white238']
+
+    homepage = "https://github.com/LLNL/axom"
+    git      = "https://github.com/LLNL/axom.git"
+
+    version('main', branch='main', submodules=True)
+    version('develop', branch='develop', submodules=True)
+    version('0.3.3', tag='v0.3.3', submodules="True")
```
Comment (ltaylor16) [inline scripts/uberenv/packages/axom/package.py:56]: Should 0.4.0 be listed in axom/package.py?

**Review by ltaylor16 — APPROVED** [review]
Looks good. But I didn't actually look at all 749 files.

File: scripts/uberenv/packages/axom/package.py (hunk 4 above)
Comment (gzagaris) [reply scripts/uberenv/packages/axom/package.py]: I guess so, although, this tag does not exist yet, but, will be created as soon as this branch gets merged -- I just added it.

**Comment by gzagaris** [comment]
> Looks good. But I didn't actually look at all 749 files.

Thanks. You shouldn't have to look at all 749 files since all these changes have already been reviewed and tested on `develop`.  

The main changes to review in this PR is to catch places where the version is not updated, as you did for the spack package and take a look at the release notes.

**Review by rhornung67 — APPROVED** [review]
Top-level files look good. I didn't look over all 749 changed files. To make rc branches easier to review, I do a PR against develop for review (since that should only have release prep changes), but have folks approve the PR to merge into master.

**Comment by gzagaris** [comment]
> Top-level files look good. I didn't look over all 749 changed files. To make rc branches easier to review, I do a PR against develop for review (since that should only have release prep changes), but have folks approve the PR to merge into master.

That makes sense. Thanks.

**Pull request merged**